core/Execute.cpp

```diff
@@ -38,10 +39,83 @@ namespace olympia
+            //const std::string tgt_name = issue_queue_topology_[i][0].substr(0, issue_queue_topology_[i][0].size()-1);
+            auto iq_node = new sparta::ResourceTreeNode(node,
+                                                        issue_queue_name,
+                                                        tgt_name,
+                                                        i,
+                                                        exe_units_string + std::string("Issue Queue"),
+                                                        &issue_queue_fact_);
+            // auto iq_resource = iq_node->getResourceAs<olympia::IssueQueue*>();
+            issue_queues_.emplace_back(iq_node);
+        }
+
     }
+    void ExecuteFactory::bindLate(sparta::TreeNode* node){
+        /*
+            For issue queue we need to establish mappings such that a mapping of target pipe to execution pipe
+            in an issue queue is known such as:
+                iq_0:
+                    "int": alu0, alu1
+                    "div": alu1
+                    "mul": alu3
+            so when we have an instruction, we can get the target pipe of an instruction and lookup available
+            execution units
+        */
+        std::unordered_map<std::string, int> pipe_to_iq_number;
+
+        for (size_t i = 0; i < issue_queues_.size(); ++i) {
+            for (auto alu_name : issue_queue_topology_[i]) {
+                for (auto& exe_pipe_tns : exe_pipe_tns_) {
+                    olympia::ExecutePipe* exe_pipe = exe_pipe_tns->getResourceAs<olympia::ExecutePipe*>();
+                    std::string exe_unit_name = exe_pipe->getName();
+
+                    if (alu_name == exe_unit_name) {
+                        olympia::IssueQueue* my_issue_queue = issue_queues_[i]->getResourceAs<olympia::IssueQueue*>();
+                        my_issue_queue->setExePipe(exe_unit_name, exe_pipe);
+                        pipe_to_iq_number[exe_unit_name] = i;
+                    }
+                }
+            }
+        }
+
+        auto setup_issue_queue_tgts = [this, &pipe_to_iq_number](sparta::TreeNode* node, std::string exe_unit) {
+            std::string topology_string = "pipe_topology_" + exe_unit + "_pipes";
+            auto pipe_topology = olympia::coreutils::getPipeTopology(node->getParent(), topology_string);
 
-    void ExecuteFactory::deleteSubtree(sparta::ResourceTreeNode*) {
+            for (size_t i = 0; i < pipe_topology.size(); ++i) {
+                for (size_t j = 0; j < pipe_topology[i].size(); ++j) {
+                    std::string pipe_name = pipe_topology[i][j];
+                    const auto tgt_pipe = InstArchInfo::execution_pipe_map.find(pipe_name);
+                    std::string exe_unit_name = exe_unit + std::to_string(i);
+
+                    int iq_num = pipe_to_iq_number.at(exe_unit_name);
```

Comment:
I'm confused -- isn't `i` the same as `iq_num`?

Reply:
`i` and `iq_num` are different. `i` is used to iterate through the pipe_topology for an execution unit, so for an `alu`, we might have `["INT", "DIV"], which we loop through those pipe targets to then set a map in issue queue of `"INT": ["ALU0", "ALU1"]. `iq_num` is used to tell us which specific issue queue has that execution unit. I've updated the comments around this code as well to explain this.